sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySchemaIOProvider.java

```diff
@@ -0,0 +1,203 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp.bigquery;
+
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.auto.service.AutoService;
+import java.io.Serializable;
+import java.util.HashMap;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Internal;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import org.apache.beam.sdk.schemas.io.SchemaIO;
+import org.apache.beam.sdk.schemas.io.SchemaIOProvider;
+import org.apache.beam.sdk.schemas.transforms.Convert;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.apache.beam.sdk.values.Row;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * An implementation of {@link SchemaIOProvider} for reading and writing to BigQuery with {@link
+ * BigQueryIO}. For a description of configuration options, see {@link
+ * BigQuerySchemaIOProvider#configurationSchema()}.
+ */
+@Internal
+@Experimental
+@AutoService(SchemaIOProvider.class)
+public class BigQuerySchemaIOProvider implements SchemaIOProvider {
+
+  /** Returns an id that uniquely represents this IO. */
+  @Override
+  public String identifier() {
+    return "bigquery";
+  }
+
+  /**
+   * Returns the expected schema of the configuration object. Note this is distinct from the schema
+   * of the data source itself. The fields are as follows:
+   *
+   * <ul>
+   *   <li>table: Nullable String - Used for reads and writes. Specifies a table to read or write
+   *       to, in the format described in {@link BigQueryHelpers#parseTableSpec}. Used as an input
+   *       to {@link BigQueryIO.TypedRead#from(String)} or {@link BigQueryIO.Write#to(String)}.
+   *   <li>query: Nullable String - Used for reads. Specifies a query to read results from using the
+   *       BigQuery Standard SQL dialect. Used as an input to {@link
+   *       BigQueryIO.TypedRead#fromQuery(String)}.
+   *   <li>queryLocation: Nullable String - Used for reads. Specifies a BigQuery geographic location
+   *       where the query job will be executed. Used as an input to {@link
+   *       BigQueryIO.TypedRead#withQueryLocation(String)}.
+   *   <li>createDisposition: Nullable String - Used for writes. Specifies whether a table should be
+   *       created if it does not exist. Valid inputs are "Never" and "IfNeeded", corresponding to
+   *       values of {@link BigQueryIO.Write.CreateDisposition}. Used as an input to {@link
+   *       BigQueryIO.Write#withCreateDisposition(BigQueryIO.Write.CreateDisposition)}.
+   * </ul>
+   */
+  @Override
+  public Schema configurationSchema() {
+    return Schema.builder()
```

Comment:
Given that this doesn't allow specifying the read method, we should document the read mode used.

Reply:
Sounds like a good idea. Done.